core/src/impl/Kokkos_Atomic_Fetch_Or.hpp

```diff
@@ -147,6 +147,26 @@ T atomic_fetch_or( volatile T * const dest_v , const T val )
 
 #endif
 #endif
+
+//----------------------------------------------------------------------------
+
+#if defined( KOKKOS_ENABLE_HIP )
+// Support for int, unsigned int, and unsigned long long int
+
+__inline__ __device__
+int atomic_fetch_or( volatile int * const dest , const int val )
+{ return atomicOr(const_cast<int*>(dest),val); }
+
+__inline__ __device__
+unsigned int atomic_fetch_or( volatile unsigned int * const dest , const unsigned int val )
+{ return atomicOr(const_cast<unsigned int*>(dest),val); }
+
+__inline__ __device__
+unsigned long long int atomic_fetch_or( volatile unsigned long long int * const dest ,
+                                         const unsigned long long int val )
+{ return atomicOr(const_cast<unsigned long long int*>(dest),val); }
```

Comment:
I almost wanna think about macros to generate the integer type supported by HIP all in one shot